tests/ui/def_id_nocore.stderr

```diff
@@ -0,0 +1,10 @@
+error: methods called `as_*` usually take self by reference or self by mutable reference; consider choosing a less ambiguous name
+  --> $DIR/def_id_nocore.rs:26:19
+   |
+LL |     pub fn as_ref(self) -> &'static str {
+   |                   ^^^^
+   |
+   = note: `-D clippy::wrong-self-convention` implied by `-D warnings`
```

Comment:
Fun fact: If you add the `&` to `self` in a `no_core` crate, rustc complains that 
```
error[E0658]: `&A` cannot be used as the type of `self` without the `arbitrary_self_types` feature
  --> tests/ui/def_id_nocore.rs:26:19
   |
26 |     pub fn as_ref(&self) -> &'static str {
   |                   ^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/44874
   = help: add `#![feature(arbitrary_self_types)]` to the crate attributes to enable
   = help: consider changing to `self`, `&self`, `&mut self`, `self: Box<Self>`, `self: Rc<Self>`, `self: Arc<Self>`, or `self: Pin<P>` (where P is one of the previous types except `Self`)
```

I don't think we need to adapt the warning though :smile: